### PR TITLE
TST: xfailed arithmetic tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/numpy_.py [hunk 1]
```diff
@@ -377,19 +387,28 @@ def arithmetic_method(self, other):
...
                 #  have an ExtensionArray, in which case we do not wrap
-                return cls(result)
+                return self._wrap_ndarray_result(result)
             return result
 
         return compat.set_function_name(arithmetic_method, f"__{op.__name__}__", cls)
 
     _create_comparison_method = _create_arithmetic_method
 
+    def _wrap_ndarray_result(self, result: np.ndarray):
```
Comment (jreback) [inline pandas/core/arrays/numpy_.py:403]: isn't this the same code you added in arrays/integer?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you add a whatsnew on this patch; this effectively enables some arithmetic on TDs that we couldn't before? or is this just going thru the correct path now?

File: pandas/core/arrays/numpy_.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/numpy_.py:403]: similar, but i dont see a nice way to share it in the short term.  medium-term ive got some thoughts kicking around

**Comment by jbrockmendel** [comment]
whatsnew added.  this fixes `IntegerArray * timedelta`

File: pandas/core/arrays/integer.py [hunk 2]
```diff
@@ -581,6 +582,12 @@ def _maybe_mask_result(self, result, mask, other, op_name: str):
             result[mask] = np.nan
             return result
 
+        if result.dtype == "timedelta64[ns]":
```
Comment (jreback) [inline pandas/core/arrays/integer.py:585]: maybe also make this a free function as you did below in numpy_?

Comment (jbrockmendel) [reply pandas/core/arrays/integer.py:585]: this is in _maybe_mask_result, which is pretty similar thematically

**Review by jreback — APPROVED** [review]

**Pull request merged**